Add config params

teadur opened the pull request:
Add some flexibility to configuration, shouldnot brake anything for users using the role

lobsterdore commented on tasks/install.yml:
```diff
@@ -19,6 +19,7 @@
   group:
     name: ssmtp
     state: present
+  when: "{{ ssmtp.restrict == true }}"
```
When clauses should be bare as far as I know eg.

```YAML
when: ssmtp.restrict == true
```

In fact you don't need the equality test here:

```YAML
when: ssmtp.restrict
```

teadur replied:
indeed, thanks for showing me how todo it correctly

lobsterdore commented on templates/ssmtp.conf.j2:
```diff
@@ -1,7 +1,20 @@
 mailhub={{ ssmtp.config.host }}:{{ ssmtp.config.port }}
 
+{% if ssmtp.config.root is defined %}
```
Can you define these settings (ssmtp.config.root, ssmtp.config.rewritedomain and the like) but set them to false in defaults? I prefer to have all possible values defined in defaults so it's obvious what options are available.

teadur replied:
Should be okey now

lobsterdore commented on defaults/main.yml:
```diff
@@ -4,7 +4,10 @@ ssmtp:
   group: ~
   user: ~
-
+  restrict: true
```
Can you alphabetically order these vars? So it should be:

```YAML
ssmtp:
  auth: true
  # User and group that will be able to use ssmtp
  group: ~
  restrict: true
  revaliases: true
  use_inventory_hostname: false
  user: ~
```

teadur replied:
Should be okey now

lobsterdore commented:
When your changes are finished please squash into one meaningful commit at well.

teadur commented:
Do i understand correctly that after squash i should make new pull request ? 
If everything is okey with you i shall do it.
For squash is comment " Allow configuration of hostname via inventory_hostname,
configuration of auth, rewritedomain, revaliases, root, fromlineoverride and ssmtp restriction." descriptive enough ?

lobsterdore commented:
@teadur no need for a new pull request, just squash your commits in your branch and that will be fine. That commit message sounds fine as well.

Commit:
Allow configuration of hostname via inventory_hostname,
configuration of auth, rewritedomain, revaliases, root, fromlineoverride and ssmtp restriction.

lobsterdore commented:
LGTM 👍
